### PR TITLE
[14.0][MIG] l10n_nl_tax_statement_icp

### PR DESCRIPTION
**Pull request by astirpe**
Depends on:

- [x] https://github.com/OCA/l10n-netherlands/pull/298

### CODE REVIEW

**Review by CasVissers-360ERP — APPROVED** [review]
LGTM, no big changes in comparison to v13.

**Comment by StefanRijnhart** [comment]
Can you rebase please?

**Comment by astirpe** [comment]
Rebased. Thanks!

**Comment by bosd** [comment]
Functional test.

The creation of an PDF ICP report throws an error:
<details>
Error:
Odoo Server Error

```
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/tools/cache.py", line 85, in lookup
    r = d[key]
  File "/.repo_requirements/odoo/odoo/tools/func.py", line 71, in wrapper
    return func(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('ir.model.data', <function IrModelData.xmlid_lookup at 0x7f12dbf947b8>, '\n            l10n_nl_tax_statement_icp.report_tax_statement_icp\n        ')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 2109, in report_download
    response = self.report_routes(reportname, docids=docids, converter=converter, context=context)
  File "/.repo_requirements/odoo/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 2044, in report_routes
    pdf = report.with_context(context)._render_qweb_pdf(docids, data=data)[0]
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_actions_report.py", line 772, in _render_qweb_pdf
    html = self_sudo.with_context(context)._render_qweb_html(res_ids, data=data)[0]
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_actions_report.py", line 812, in _render_qweb_html
    return self._render_template(self.sudo().report_name, data), 'html'
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_actions_report.py", line 580, in _render_template
    return view_obj._render_template(template, values)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_ui_view.py", line 1709, in _render_template
    return self.browse(self.get_view_id(template))._render(values, engine)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_ui_view.py", line 1604, in get_view_id
    return view and view.id or self.env['ir.model.data'].xmlid_to_res_id(template, raise_if_not_found=True)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_model.py", line 1894, in xmlid_to_res_id
    return self.xmlid_to_res_model_res_id(xmlid, raise_if_not_found)[1]
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_model.py", line 1885, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(xmlid)[1:3]
  File "<decorator-gen-36>", line 2, in xmlid_lookup
  File "/.repo_requirements/odoo/odoo/tools/cache.py", line 90, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_model.py", line 1878, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % xmlid)
ValueError: External ID not found in the system: 
            l10n_nl_tax_statement_icp.report_tax_statement_icp

            ```

**Comment by bosd** [comment]
Keep up the good work!
I do have some remarks from my functional test of this module.

Can you enhance the description for setting up this module? as is not very clear.

> Follow the configuration steps as described in l10n_nl_tax_statement and set the tag 3b omzet needed for this report.

The l10n_nl_tax_statement says:

>The user must belong to the Show Full Accounting Features group, to be able to access the Invoicing -> Configuration -> Accounting -> NL BTW Tags menu.

I have full accounting features access. There's no NL BTW Tags in the menu, just taxes.
That screen looks completely different from the V12 version.

Well, the module works ut of the box. anyway :+1: 

Whilst creating an invoice with services, an error occured:
<details>
``

Fout:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 347, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1377, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1369, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 396, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 383, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/build/OCA/l10n-netherlands/setup/l10n_nl_tax_statement/odoo/addons/l10n_nl_tax_statement/models/account_move.py", line 34, in write
    return super().write(values)
  File "/.repo_requirements/odoo/addons/account/models/account_move.py", line 1862, in write
    res = super(AccountMove, self.with_context(check_move_validity=False, skip_account_move_synchronization=True)).write(vals)
  File "/.repo_requirements/odoo/addons/mail/models/mail_thread.py", line 322, in write
    result = super(MailThread, self).write(values)
  File "/.repo_requirements/odoo/addons/mail/models/mail_activity.py", line 766, in write
    return super(MailActivityMixin, self).write(vals)
  File "/.repo_requirements/odoo/odoo/models.py", line 3666, in write
    field.write(self, vals[fname])
  File "/.repo_requirements/odoo/odoo/fields.py", line 3043, in write
    return self.write_batch([(records, value)])
  File "/.repo_requirements/odoo/odoo/fields.py", line 3064, in write_batch
    return self.write_real(records_commands_list, create)
  File "/.repo_requirements/odoo/odoo/fields.py", line 3215, in write_real
    comodel.browse(command[1]).write(command[2])
  File "/home/odoo/build/OCA/l10n-netherlands/setup/l10n_nl_tax_statement/odoo/addons/l10n_nl_tax_statement/models/account_move_line.py", line 36, in write
    self._l10n_nl_vat_statement_check_state()
  File "/home/odoo/build/OCA/l10n-netherlands/setup/l10n_nl_tax_statement/odoo/addons/l10n_nl_tax_statement/models/account_move_line.py", line 56, in _l10n_nl_vat_statement_check_state
    % (line.l10n_nl_vat_statement_id.name,)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: not all arguments converted during string formatting


``


</details>


During the functional test. Services turnover was stated on the ICP report as turnover on goods.
This seems to be an issue with the current COA.

Steps to reproduce:
Create an new service product.
Set customer taxes to “Verkopen/omzet hoog diensten”.
Add it to an invoice to an partner in another EU country.
Result: the selected account is “800200 Omzet binnen EU Handelsgoederen 1”.

Set customer taxes to “Verkopen/omzet hoog diensten”.
Add it to an invoice to an partner in NL.
Result: the selected account is “80100 Omzet binnen NL handelsgoederen 1”.

The expected result was an selected chart of account in the 8003 range, just like it was in the V8 days.
Which included account: ```8003080 Omzet diensten in landen binnen EU```
https://github.com/OCA/l10n-netherlands/blob/649347ac6472004860f3108d54f5259372666872/l10n_nl_chart_rgs/data/account_account_template.xml#L21091-L21096

Sadly that chart is no longer available.
The current COA only allows us to sell ``handelsgoederen binnen de EU`` :wink: followed by a ``1,2,3``.
(not sure where the 1,2,3, stands for or the distinction to make between them)

On which scheme the current COA of odoo is based??


Have'nt seen the logs yet.
But even on a tiny database with some small amount of demo data it takes very long to load the statements.

File: l10n_nl_tax_statement_icp/views/report_tax_statement.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2018-2020 Onestein (<https://www.onestein.eu>)
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="action_report_tax_statement_icp" model="ir.actions.report">
+        <field name="name">NL ICP Statement</field>
+        <field name="model">l10n.nl.vat.statement</field>
+        <field name="report_type">qweb-pdf</field>
+        <field name="report_name">
+            l10n_nl_tax_statement_icp.report_tax_statement_icp
```
Comment (CasVissers-360ERP) [inline l10n_nl_tax_statement_icp/views/report_tax_statement.xml]: The spaces here results in a traceback:
ValueError: External ID not found in the system:
l10n_nl_tax_statement_icp.report_tax_statement_icp

Comment (NL66278) [reply l10n_nl_tax_statement_icp/views/report_tax_statement.xml]: @CasVissers-360ERP I can confirm the exception. @astirpe Do you intend to take this up?

Comment (NL66278) [reply l10n_nl_tax_statement_icp/views/report_tax_statement.xml]: After solving the report error, I do not see the customer lines in the generated report. Researching further...

Comment (NL66278) [reply l10n_nl_tax_statement_icp/views/report_tax_statement.xml]: Aha, I first need to post the statement...

Comment (NL66278) [reply l10n_nl_tax_statement_icp/views/report_tax_statement.xml]: @astirpe will you take up the comment from @CasVissers-360ERP - confirmed by me - on the spaces in the xml causing an error?

**Comment by bosd** [comment]
What is the current status? Is this one ready for review?

**Comment by astirpe** [comment]
Ready for review. Sorry for the delay

**Review by NL66278 — APPROVED** [review]
:+1: LGTM

**Comment by bosd** [comment]
It is oca review day ::smile::
 Reviews on this one are approved, but it still is not merging. :(


Can someone provide feedback on my issue/question that with the new chart of accounts from v8 and above it seems impossible to use "
Verkopen/omzet hoog diensten" on this statement.

Should I open seperate issue for this??

**Comment by StefanRijnhart** [comment]
@bosd I think your question is out of scope for this PR. But you can check out the comments on https://github.com/OCA/l10n-netherlands/pull/328/files (which was my workaround for the same issue).

/ocabot merge nobump

**Pull request merged**